src/compas/datastructures/network/core/graph.py

```diff
@@ -1275,8 +1275,6 @@ def neighborhood(self, key, ring=1):
                 temp += self.neighbors(key)
             nbrs.update(temp)
             i += 1
-        if key in nbrs:
-            nbrs.remove(key)
```

Comment:
this line does not remove potential duplicates, but rather the original node which should not appear in the list of its own neighbours...

however, there is another problem related to that that is indeed an actual bug (will let you figure it out :)

Reply:
the change should actually be
```python
def neighborhood(self, vertex, ring=1):
    nbrs = set(self.neighbors(vertex))
    ...
    if vertex in nbrs:
        nbrs.remove(vertex)
```

Reply:
@duchaoyu ping

Reply:
please fix so i can merge...